### PR TITLE
Adds: delete edge-app command - same as screen deletion

### PR DESCRIPTION
**Pull request by korvyashka**
Refs https://phorge.wireload.net/T7415

Same as screen deletion - ask for name promt before deletion

`screenly edge-app delete`

Also updated manifest to have app_id optional -> `Option<String>`

### CODE REVIEW

**Review by vpetersson — COMMENTED** [review]
What happens to the files on disk? Do we wipe the app id from the yaml file?

**Comment by korvyashka** [comment]
@vpetersson discussed with @sergey-borovkov and decided not to do it.
Let me know if we actually want to do this.

**Comment by vpetersson** [comment]
What's the rational? We currently don't enforce app name to be unique, right?

**Comment by vpetersson** [comment]
Right so then we can't use this approach. We much change it to use the app id. Id say @MuratUrsavas suggestion plus app id seems reasonable.

**Comment by korvyashka** [comment]
@vpetersson I'm not sure I'm following.

- suggestion from @MuratUrsavas was in another PR, https://github.com/Screenly/cli/pull/95 and it is cause PR description was wrong - update uses '--name=' of course not 'name=', so it is not relevant.

- Deletion already happens by app_id - either '--app_id=' or if not passed, from manifest in current folder. Name input is just to confirm deletion and not allow it happen instantaneusly when calling delete command for instance accidently.

- So only question left is if we need to delete screenly.yml on delete or delete app_id there or delete whole folder.
@sergey-borovkov said we shouldn't, I don't have strong opinion here - let me know if we should.

**Comment by sergey-borovkov** [comment]
> Right so then we can't use this approach. We much change it to use the app id. Id say @MuratUrsavas suggestion plus app id seems reasonable.

The name is just for confirmation. You delete with id.

I don't think we should remove anything from the disk. Makes no sense to me to be honest.

File: src/cli.rs [hunk 1]
```diff
@@ -299,6 +299,16 @@ pub enum EdgeAppCommands {
         #[arg(short, long)]
         path: Option<String>,
 
+        /// Edge app id. If not specified CLI will use the id from the manifest.
+        #[arg(short, long)]
+        app_id: Option<String>,
+    },
+    /// Deletes an edge app. This cannot be undone.
```
Comment (sergey-borovkov) [inline src/cli.rs:306]: Please capitalize. Edge App. Everywhere. If you see it incorrectly nearby please ifx it as well

File: src/cli.rs [hunk 2]
```diff
@@ -299,6 +299,16 @@ pub enum EdgeAppCommands {
...
+    Delete {
+        /// Path to the directory with the manifest. If not specified CLI will use the current working directory.
+        #[arg(short, long)]
+        path: Option<String>,
+
         /// Edge app id. If not specified CLI will use the id from the manifest.
```
Comment (sergey-borovkov) [inline src/cli.rs:312]: Like here

File: src/cli.rs [hunk 3]
```diff
@@ -299,6 +299,16 @@ pub enum EdgeAppCommands {
         #[arg(short, long)]
         path: Option<String>,
 
+        /// Edge app id. If not specified CLI will use the id from the manifest.
+        #[arg(short, long)]
+        app_id: Option<String>,
+    },
+    /// Deletes an edge app. This cannot be undone.
+    Delete {
```
Comment (sergey-borovkov) [inline src/cli.rs:307]: Why is it not taking id optionally?

File: src/cli.rs [hunk 4]
```diff
@@ -946,6 +956,34 @@ pub fn handle_cli_edge_app_command(command: &EdgeAppCommands) {
                 }
             }
         },
+        EdgeAppCommands::Delete { path, app_id } => {
+            let actual_app_id = get_actual_app_id(app_id, path);
+            match edge_app_command.get_app_name(&actual_app_id) {
+                Ok(name) => {
+                    info!("You are about to delete the app named \"{}\".  This operation cannot be reversed.", name);
```
Comment (sergey-borovkov) [inline src/cli.rs:963]: Edge App named maybe here as well for consistency?

**Review by sergey-borovkov — CHANGES_REQUESTED** [review]

File: src/cli.rs (hunk 3 above)
Comment (korvyashka) [reply src/cli.rs:307]: /// Edge App id. If not specified CLI will use the id from the manifest.
        #[arg(short, long)]
        app_id: Option<String>,

Comment (korvyashka) [reply src/cli.rs:307]: It is

Comment (sergey-borovkov) [reply src/cli.rs:350]: my bad, it's because how diff shows - did not notice

**Review by sergey-borovkov — APPROVED** [review]

**Pull request merged**